src/delay_cycles.rs

```diff
@@ -103,11 +116,7 @@ const fn select(info: Cycles, cycles: u64, above: u64) -> Selection {
 }
 
 impl<const CYCLES: u64, const MUL: u64, const DIV: u64> Delayer<CYCLES, MUL, DIV> {
-    // Multiply first to avoid precision loss.
-    // With a u64 there is no overflow when MUL is lower than:
-    //  (2^64-1)/25_769_803_784 == 715_827_882.
-    // Since MUL is usually CPU_FREQUENCY_HZ, this allows up to 715.83 MHz.
-    const TOTAL_CYCLES: u64 = CYCLES * MUL / DIV;
+    const TOTAL_CYCLES: u64 = compute_total_cycles(CYCLES, MUL, DIV);
```

Comment:
I guess you can simply type the code here within an experession block. No need for a one time use function. Unless that is limited by Rust of course.